mep/accounts/management/commands/twitterbot_100years.py

```diff
@@ -0,0 +1,264 @@
+        # auth = tweepy.OAuthHandler('', '')
+        # auth.set_access_token('', '')
+        # api = tweepy.API(auth)
+        # api.update_status(content)
+
+
+tweetable_event_types = ['Subscription', 'Renewal', 'Reimbursement',
+                         'Borrow', 'Purchase', 'Request']
+
+
+def can_tweet(ev, date):
+    '''check if the event can be tweeted on the specified day'''
+
+    # convert to string and compare against partial dates
+    # to ensure we don't tweet an event with an unknown date
+    date = date.isoformat()
+
+    # disallows subscription on start date != purchase
+    if ev.event_label == 'Subscription' and \
+       ev.subscription.partial_purchase_date != date:
+        return False
+
+    return any([
+        # subscription event and date matches purchase
+        (ev.event_label == 'Subscription' and
+         ev.subscription.partial_purchase_date == date),
+        # borrow event and date matches end date
+        (ev.event_label == 'Borrow' and ev.partial_end_date == date),
+        # any other tweetable event and date matches start
+        ev.event_label in tweetable_event_types and
+        ev.partial_start_date == date])
+
+
+def tweet_content(ev, event_date):
+    # handle multiple members, but use first member for url
+    member = ev.account.persons.first()
+
+    if isinstance(event_date, str):
+        try:
+            event_date = datetime.date(*[int(n) for n in event_date.split('-')])
+        except TypeError:
+            # given a partial date
+            return
+
+    member_name = ' and '.join(m.name for m in ev.account.persons.all())
+    prolog = '#100YearsAgoToday on %s:' % \
+        event_date.strftime(Command.date_format)
+    content = None
+
+    event_label = ev.event_label
+
+    if event_label == 'Subscription' and \
+       ev.subscription.purchase_date == event_date:
+        content = '%s joined the Shakespeare and Company lending library.' % \
+            member_name
+
+    elif event_label in ['Borrow', 'Purchase', 'Request']:
+        work = work_label(ev.work)
+        # convert event type into verb for the sentence
+        verb = '%sed' % ev.event_type.lower().rstrip('e')
+        if event_label == 'Borrow' and ev.end_date == event_date:
+            verb = 'returned'
+        content = '%s %s %s.' % (member_name, verb, work)
+
+    # renewal
+    elif event_label == 'Renewal' and ev.subscription.purchase_date == event_date:
+        # renewed for 2 months at 1 volume per month
+        content = '%s renewed for %s' % \
+            (member_name, ev.subscription.readable_duration())
+        # include volume count if known
+        if ev.subscription.volumes:
+            content = '%s at %d volume%s per month.' % \
+                (content, ev.subscription.volumes,
+                 '' if ev.subscription.volumes == 1 else 's')
+
+    elif event_label == 'Reimbursement':
+        # renewed for 2 months at 1 volume per month
```

Comment:
think this comment was copy-pasted from above but not edited to match the reimbursement message format

Reply:
ha, thanks, yes